xarray/conventions.py

```diff
@@ -958,73 +460,22 @@ def decode_cf_variable(name, var, concat_characters=True, mask_and_scale=True,
         if string_encoding is not None:
             data = BytesToStringArray(data, string_encoding)
 
-    unsigned = pop_to(attributes, encoding, '_Unsigned')
-    if unsigned and mask_and_scale:
-        if data.dtype.kind == 'i':
-            data = UnsignedIntTypeArray(data)
-        else:
-            warnings.warn("variable %r has _Unsigned attribute but is not "
-                          "of integer type. Ignoring attribute." % name,
-                          SerializationWarning, stacklevel=3)
+    # TODO(shoyer): convert everything above to use coders
+    var = Variable(dimensions, data, attributes, encoding)
 
     if mask_and_scale:
-        if 'missing_value' in attributes:
-            # missing_value is deprecated, but we still want to support it as
-            # an alias for _FillValue.
-            if ('_FillValue' in attributes and
-                not utils.equivalent(attributes['_FillValue'],
-                                     attributes['missing_value'])):
-                raise ValueError("Conflicting _FillValue and missing_value "
-                                 "attributes on a variable {!r}: {} vs. {}\n\n"
-                                 "Consider opening the offending dataset "
-                                 "using decode_cf=False, correcting the "
-                                 "attributes and decoding explicitly using "
-                                 "xarray.decode_cf()."
-                                 .format(name, attributes['_FillValue'],
-                                         attributes['missing_value']))
-            attributes['_FillValue'] = attributes.pop('missing_value')
-
-        fill_value = pop_to(attributes, encoding, '_FillValue')
-        if isinstance(fill_value, np.ndarray) and fill_value.size > 1:
-            warnings.warn("variable {!r} has multiple fill values {}, "
-                          "decoding all values to NaN."
-                          .format(name, fill_value),
-                          SerializationWarning, stacklevel=3)
-
-        scale_factor = pop_to(attributes, encoding, 'scale_factor')
-        add_offset = pop_to(attributes, encoding, 'add_offset')
-        has_fill = (fill_value is not None and
-                    not np.any(pd.isnull(fill_value)))
-        if (has_fill or scale_factor is not None or add_offset is not None):
-            if has_fill and np.array(fill_value).dtype.kind in ['U', 'S', 'O']:
-                if string_encoding is not None:
```

Comment:
I removed this check. I think I put this logic in the wrong place: we really should ensure that `_FillValue` is not provided when writing/encoding a variable to disk, not in reading/decoding.